```
error[E0412]: cannot find type `Rc` in this scope
  --> ports/libsimpleservo/api/src/gl_glue.rs:82:29
   |
82 |     pub fn init() -> Result<Rc<Gl>, &'static str> {
   |                             ^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
82 |     use std::rc::Rc;
   |
error[E0412]: cannot find type `Gl` in this scope
  --> ports/libsimpleservo/api/src/gl_glue.rs:82:32
   |
82 |     pub fn init() -> Result<Rc<Gl>, &'static str> {
   |                                ^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
   |
82 |     use servo::gl::Gl;
   |
error: aborting due to 2 previous errors

```

--- a/ports/libsimpleservo/api/Cargo.toml
+++ b/ports/libsimpleservo/api/Cargo.toml
@@ -14,6 +14,9 @@ serde_json = "1.0"
 [target.'cfg(not(target_os = "macos"))'.dependencies]
 libc = "0.2"
 
+[target.'cfg(target_os = "macos")'.dependencies]
+core-foundation = "0.6"
+
 [target.'cfg(target_os = "windows")'.dependencies]
 winapi = "0.3.2"
 

--- a/ports/libsimpleservo/api/src/gl_glue.rs
+++ b/ports/libsimpleservo/api/src/gl_glue.rs
@@ -2,9 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-use std::rc::Rc;
-
-pub type ServoGl = Rc<dyn servo::gl::Gl>;
+pub type ServoGl = std::rc::Rc<dyn servo::gl::Gl>;
 
 #[cfg(any(target_os = "android", target_os = "windows"))]
 #[allow(non_camel_case_types)]
@@ -56,7 +54,7 @@ pub mod egl {
     }
 
     #[cfg(target_os = "windows")]
-    pub fn init() -> Result<Rc<Gl>, &'static str> {
+    pub fn init() -> Result<crate::gl_glue::ServoGl, &'static str> {
         info!("Loading EGL...");
 
         let dll = b"libEGL.dll\0" as &[u8];
@@ -77,10 +75,38 @@ pub mod egl {
     }
 }
 
-#[cfg(any(target_os = "windows", target_os = "linux", target_os = "macos"))]
+#[cfg(any(target_os = "windows", target_os = "linux"))]
 pub mod gl {
     pub fn init() -> Result<crate::gl_glue::ServoGl, &'static str> {
-        // FIXME: Add an OpenGL version
-        unimplemented!()
+        unimplemented!();
+    }
+}
+
+#[cfg(target_os = "macos")]
+pub mod gl {
+    use core_foundation::base::TCFType;
+    use core_foundation::bundle::{
+        CFBundleGetBundleWithIdentifier, CFBundleGetFunctionPointerForName,
+    };
+    use core_foundation::string::CFString;
+    use servo::gl::GlFns;
+    use std::os::raw::c_void;
+    use std::str;
+
+    pub fn init() -> Result<crate::gl_glue::ServoGl, &'static str> {
+        info!("Loading OpenGL...");
+        let gl = unsafe {
+            GlFns::load_with(|addr| {
+                let symbol_name: CFString = str::FromStr::from_str(addr).unwrap();
+                let framework_name: CFString = str::FromStr::from_str("com.apple.opengl").unwrap();
+                let framework =
+                    CFBundleGetBundleWithIdentifier(framework_name.as_concrete_TypeRef());
+                let symbol =
+                    CFBundleGetFunctionPointerForName(framework, symbol_name.as_concrete_TypeRef());
+                symbol as *const c_void
+            })
+        };
+        info!("OpenGL loaded");
+        Ok(gl)
     }
 }

--- a/ports/libsimpleservo/capi/Cargo.toml
+++ b/ports/libsimpleservo/capi/Cargo.toml
@@ -15,6 +15,7 @@ bench = false
 [dependencies]
 simpleservo = { path = "../api" }
 log = "0.4"
+env_logger = "0.6"
 
 [features]
 default = ["unstable", "default-except-unstable"]

--- a/ports/libsimpleservo/capi/src/lib.rs
+++ b/ports/libsimpleservo/capi/src/lib.rs
@@ -5,6 +5,7 @@
 #[macro_use]
 extern crate log;
 
+use env_logger;
 use simpleservo::{self, gl_glue, EventLoopWaker, HostTrait, InitOptions, ServoGlue, SERVO};
 use std::ffi::{CStr, CString};
 use std::mem;
@@ -68,6 +69,8 @@ fn init(
     wakeup: extern "C" fn(),
     callbacks: CHostCallbacks,
 ) {
+    crate::env_logger::init();
+
     let args = unsafe { CStr::from_ptr(opts.args) };
     let args = args.to_str().map(|s| s.to_string()).ok();
 

